src/odemis/acq/align/autofocus.py

```diff
@@ -114,6 +115,28 @@ def MeasureOpticalFocus(image):
     return cv2.Laplacian(image, cv2.CV_64F).var()
 
 
+def Measure1d(image):
+    """
+    Given an image of a 1 line ccd, measure the focus based on the inverse of the width of a gaussian fit of the data.
+    It is assumed that the signal is in focus when the width of the signal is smallest and therefore sigma is smallest.
+    image (model.DataArray): 1D image from 1 line ccd.
+    returns (float): The focus level of the image.
+    """
+    def gauss(x, *p):
+        a, b, c, d = p
```

Comment:
Isn't it possible to just write `def gauss(x, a,b,c,d):`?
Also, "a,b,c,d" are really obfuscated names. I'd suggest to use the same names as in `util.peak.GaussianFit()`: pos, width, amplitude, offset (not sure of the order though).
It might also be interesting to compare your formula to the one used in `GaussianFit()` and check if they can learn from each other...
At least, it should be possible to drop the `.`, and replace `numpy.power(..., 2) by `(...)**2`.

Reply:
Updated the pull request, removed the *p. I think the biggest difference between the two functions is the base offset I'm using. That is necessary here because the base can be at 800 or something and the curve fit returns garbage without that offset, I don't know if that would improve the other function.